_posts/2025-06-24-kafka-queue.md

```diff
@@ -0,0 +1,703 @@
+
+The share group does not support exactly-once semantics and does not provide ordering guarantee as multiple consumers can fetch records from the same partition. However, these features may potentially be added in the future. The dead letter queue is currently not supported either but there is already an open [KIP](https://cwiki.apache.org/confluence/display/KAFKA/KIP-1191%3A+Dead-letter+queues+for+share+groups) for it.
+
+### Comparing share and consumer groups with Strimzi
+
+I have set up a single node cluster on my machine using Strimzi’s [Quickstart](https://strimzi.io/quickstarts/) and created a topic called `kafka-queue-topic` with 2 partitions:
+```sh
+$ cat << EOF | kubectl create -n kafka -f -
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaTopic
+metadata:
+  name: kafka-queue-topic
+  labels:
+    strimzi.io/cluster: "my-cluster"
+spec:
+  partitions: 2
+  replicas: 1
+EOF
+```
+
+Since this feature is still in early access, as of Apache Kafka 4.0.0, you need to explicitly enable it by setting the following:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  # ...
+spec:
+  kafka:
+    config:
+      # ...
+      group.coordinator.rebalance.protocols: classic,consumer,share
+      unstable.api.versions.enable: true
+      share.coordinator.state.topic.replication.factor: 1
+    # ...
+    version: 4.0.0
+    metadataVersion: 4.0-IV3
+  # ...
+```
+Since I'm running a single node cluster, `share.coordinator.state.topic.replication.factor` also had to be set to 1. We will cover more about this topic later.
+
+Using the Kafka command line tools, I will produce some records on the topic and consume those records using both consumer and share groups and show how they behave differently. 
+
+In this example, I will start with 3 share consumers, all joining the same share group called, `share-group` and then 3 regular consumers, also all joining the same consumer group called, "consumer-group". I will run the following command in different terminal windows with different pod names, e.g.  kafka-share-consumer-0,  kafka-share-consumer-1, kafka-share-consumer-2:
+```sh
+$ kubectl -n kafka run kafka-share-consumer-0 -ti --image=quay.io/strimzi/kafka:0.46.0-kafka-4.0.0 --rm=true --restart=Never \
```

Comment:
Maybe we can bump the image version here and in all the other cases to 0.47.0 to use the latest and greatest?

(You actually use 0.47.0 in some of the commands later which makes it even ore confusing)